src/components/ec-inline-input-field/components/edit/edit.vue

```diff
@@ -0,0 +1,118 @@
+<template>
+  <div
+    class="tw-w-full tw-flex"
+    data-test="ec-inline-input-field-edit"
+  >
+    <ec-input-field
+      ref="input"
+      v-model="value"
+      type="text"
+      data-test="ec-inline-input-field-edit__input"
+      @keydown.enter="submitViaKeyboard"
+      @keydown.esc="cancelViaKeyboard"
+    />
+    <div class="tw-flex tw-ml-20 tw-mt-12">
+      <button
+        class="ec-inline-input-field-edit__action"
+        data-test="ec-inline-input-field-edit__submit-action"
+        @click="submit"
+        @keydown.enter.space.prevent="submitViaKeyboard"
+      >
+        <ec-icon
+          class="ec-inline-input-field-edit__action-icon"
+          name="simple-check"
+          :size="16"
+        />
+      </button>
+      <button
+        class="ec-inline-input-field-edit__action tw-ml-8"
+        data-test="ec-inline-input-field-edit__cancel-action"
+        @click="cancel"
+        @keydown.enter.space.prevent="cancelViaKeyboard"
+      >
+        <ec-icon
+          class="ec-inline-input-field-edit__action-icon"
+          name="simple-close"
+          :size="16"
+        />
+      </button>
+    </div>
+  </div>
+</template>
+
+<script>
+import { EDITING, LOADING, READ_ONLY } from '@/enums/input-status';
+import EcIcon from '@/components/ec-icon';
+import EcInputField from '@/components/ec-input-field';
+
+export default {
+  name: 'EcInlineInputFieldEdit',
+  components: { EcIcon, EcInputField },
+  props: {
+    originalValue: {
+      default: '',
+      type: String,
+    },
+    status: {
+      type: String,
+      default: READ_ONLY,
+      validator(value) {
+        return [READ_ONLY, EDITING, LOADING].includes(value);
+      },
+    },
+  },
+  data() {
+    return {
+      value: null,
+    };
+  },
+  watch: {
+    status: {
+      immediate: true,
+      handler() {
+        this.value = this.originalValue;
+        this.$nextTick(() => {
+          this.$refs.input.$el.querySelector('input').focus();
```

Comment:
Second though here, why are you focusing the input every time the status changes? If submitting takes 30 seconds, user might be doing something else. After submitting is done, parent will change the status from LOADING to IS_EDITING and it will re-focus the input. I'm not sure that's right.

Reply:
Each time the `status` was changed (and this was set to a different value than `editing`) the component was removed (due to the `v-if` directive in the parent component), so it should only be entered here when the component is re-mounted. But the truth is that it was a horrible solution, taking into account that we already have a hook precisely for that purpose. I have taken that piece of code to the "mounted" method.